pyomo/contrib/gdpopt/nlp_solve.py

```diff
@@ -311,19 +312,17 @@ def detect_unfixed_discrete_vars(model):
 
 def preprocess_subproblem(m, config):
     """Applies preprocessing transformations to the model."""
-    # fbbt(m, integer_tol=config.integer_tolerance)
+    # First do FBBT
+    fbbt(m, integer_tol=config.integer_tolerance,
+         feasibility_tol=config.constraint_tolerance)
     xfrm = TransformationFactory
-    xfrm('contrib.propagate_eq_var_bounds').apply_to(m)
+    # Now that we've tightened bounds, see if any variables are fixed because
+    # their lb is equal to the ub (within tolerance)
     xfrm('contrib.detect_fixed_vars').apply_to(
-        m, tolerance=config.variable_tolerance)
-    xfrm('contrib.propagate_fixed_vars').apply_to(m)
+         m, tolerance=config.variable_tolerance)
+    # Now, if something got fixed to 0, we might have 0*var terms to remove
     xfrm('contrib.remove_zero_terms').apply_to(m)
-    xfrm('contrib.propagate_zero_sum').apply_to(m)
-    xfrm('contrib.constraints_to_var_bounds').apply_to(
-        m, tolerance=config.variable_tolerance)
-    xfrm('contrib.detect_fixed_vars').apply_to(
-        m, tolerance=config.variable_tolerance)
-    xfrm('contrib.propagate_zero_sum').apply_to(m)
+    # Last, check if any constraints are now trivial and deactivate them
     xfrm('contrib.deactivate_trivial_constraints').apply_to(
         m, tolerance=config.constraint_tolerance)
```

Comment:
If the `deactivate_trivial_constraints` transformation does what I think, then it should also be covered by FBBT by using the `deactivate_satisfied_constraints` option in the call to `fbbt`. It is worth double checking that these do the same thing though.

Reply:
Ah, good point. They do indeed do the same thing. However, I guess you could get more trivially satisfied constraints after the `detect_fixed_vars` transformation, so maybe this call is still worth it?